packages/cli/lib/lib/webpack/create-load-manifest.js

```diff
@@ -1,14 +1,18 @@
-module.exports = assets => {
+module.exports = (assets, isESMBuild) => {
   let mainJs, mainCss, scripts=[], styles=[];
   for (let filename in assets) {
     if (!/\.map$/.test(filename)) {
-      if (/route-/.test(filename)) {
+      if (/route-/.test(filename) && !isESMBuild) {
+        scripts.push(filename);
+      } else if (/route-(.+)\.esm\.js/.test(filename) && isESMBuild) {
         scripts.push(filename);
       } else if (/chunk\.(.+)\.css$/.test(filename)) {
         styles.push(filename);
       } else if (/^bundle(.+)\.css$/.test(filename)) {
         mainCss = filename;
-      } else if (/^bundle(.+)\.js$/.test(filename)) {
+      } else if (!isESMBuild && /^bundle(.+)\.js$/.test(filename)) {
+        mainJs = filename;
+      } else if (isESMBuild && /bundle\.\w{5}\.esm\.js/.test(filename)) {
```

Comment:
Such a crazy if statement we got here lol

Reply:
heh.. i am not the best regexp maker out thr 😂

Reply:
For this block & the `route-` one above it, it probably cleaner to do something like this instead:

```js
let isESM = x => /\.esm\.js$/.test(x);
let isMatch = (x, y) => isESM(x) === y;

//---

      if (/route-/.test(filename)) {
        // both ESM & regular match here
        isMatch(filename, isESMBuild) && scripts.push(filename);
      } 
      // ...
      else if (/^bundle(.+)\.js$/.test(filename)) {
        // both bundles match here
        if (isMatch(filename, isESMBuild)) {
          mainJs = filename;
        }
      }
```

Reply:
done. Thanks for the suggestion